query/promql/promqltests/README.md

```diff
@@ -0,0 +1,11 @@
+# PromQL End-to-end Tests 
+
+The code under this directory is organized in the go submodule `github.com/influxdata/promqltests`.
+This was made as a go submodule to avoid bringing unnecessary dependencies into InfluxDB.
+This module is _not_ intended to be exported and used by other modules.
```

Comment:
I don't remember the details of what went wrong last time we tried to use multiple modules in a closed source repo, so I'm a little wary of this.

Maybe this should have import path `github.com/influxdata/influxdb/query/internal/promqltests`, or something similar with `internal`? That way the go tooling will properly enforce that the module can't be used outside of `github.com/influxdata/influxdb/query/...`.

Possibly another option would be [an empty go.mod](https://github.com/golang/go/wiki/Modules#can-an-additional-gomod-exclude-unnecessary-content-do-modules-have-the-equivalent-of-a-gitignore-file
):

> An empty go.mod in a directory will cause that directory and all of its subdirectories to be excluded from the top-level Go module.

I'm not sure if the `make test-promql-e2e` action would behave as intended that way.

Reply:
I see your point with `internal`. I guess that's a good way of prescribing imports.

I think I want to prevent to things:
 1. InfluxDB shouldn't import this --> solved with `internal`
 2. the module shouldn't be used outside.

I think that in the README I was addressing `2`. I don't know if `internal` would be enough. The submodule is now `github.com/influxdata/promqltests`. I'll see if changing to `github.com/influxdata/internal/promqltests` changes things.